test/cpp/jit/test_gpu_tensorcore.cpp

```diff
@@ -891,6 +891,1464 @@ TEST_F(NVFuserTest, FusionVoltaMatMulNT_CUDA) {
+  // [Mo,Ko,M,K]
+  tv0cw->merge(-2);
+  tv0r->merge(-2);
+  scheduler_utils::matmul_utils::scheduleContiguousVectorLoad(
+      tv0cw, gemm_tile1, 8);
+  scheduler_utils::matmul_utils::scheduleContiguousVectorLoad(
+      tv0r, gemm_tile1, 8);
+  tv0cw->setMemoryType(MemoryType::Shared);
+  // [Mo,Ko,i,wy,wx,v]
+
+  // [No,Ko,N,K]
+  tv1cw->merge(-2);
+  tv1r->merge(-2);
+  // [No,Ko,i,wy,wx,v]
+  scheduler_utils::matmul_utils::scheduleContiguousVectorLoad(
+      tv1cw, gemm_tile1, 8);
+  scheduler_utils::matmul_utils::scheduleContiguousVectorLoad(
+      tv1r, gemm_tile1, 8);
+  tv1cw->setMemoryType(MemoryType::Shared);
+
+  // Schedule mma input
+  // ---------------------------------------------------------------------------
+  tv0cr->applyMmaSwizzle(mma_builder1.operand(MmaOptions::Operand::A).build());
+  // [... Mi, Ni, Ki] want [Ni, Mi, Ki]
+  tv0b->reorder({{-2, -3}, {-3, -2}});
+  tv0b->applyMmaSwizzle(mma_builder1.operand(MmaOptions::Operand::A).build());
+
+  tv1cr->applyMmaSwizzle(mma_builder1.operand(MmaOptions::Operand::B).build());
+  tv1b->applyMmaSwizzle(mma_builder1.operand(MmaOptions::Operand::B).build());
+
+  // Schedule mma output
+  // ---------------------------------------------------------------------------
+  tv3c->applyMmaSwizzle(
+      mma_builder1.operand(MmaOptions::Operand::NotOperand).build());
+  tv3cw->applyMmaSwizzle(
+      mma_builder1.operand(MmaOptions::Operand::NotOperand).build());
+  tv3h->applyMmaSwizzle(
+      mma_builder1.operand(MmaOptions::Operand::NotOperand).build());
+  tv3->applyMmaSwizzle(
+      mma_builder1.operand(MmaOptions::Operand::NotOperand).build());
+  tv3cw->setMemoryType(MemoryType::Shared);
+
+  // Parallelize
+  //  0  1  2   3   4   5  6  7
+  // [Mo No Mwo Nwo Mw Nw (Mi Ni)]
+  // Gemm 1
+  tv3c->axis(3)->parallelize(ParallelType::TIDz);
+  tv3c->axis(4)->parallelize(ParallelType::TIDy);
+
+  tv3->computeAt(tv3cw, -2);
+  tv3cw->axis(2)->parallelize(ParallelType::TIDz);
+  tv3cw->axis(3)->parallelize(ParallelType::TIDy);
+
+  // Gemm 2
+  tv4->axis(2)->parallelize(ParallelType::TIDz);
+  tv4->axis(3)->parallelize(ParallelType::TIDy);
+  tv4c->axis(3)->parallelize(ParallelType::TIDz);
+  tv4c->axis(4)->parallelize(ParallelType::TIDy);
+
+  tv4->axis(0)->parallelize(ParallelType::BIDx);
+  tv4->axis(1)->parallelize(ParallelType::BIDy);
+
+  auto options = at::TensorOptions().dtype(at::kHalf).device(at::kCUDA, 0);
+  auto t0 = at::randn({M, K1}, options);
+  auto t1 = at::randn({K2, K1}, options);
+  auto t2 = at::randn({N, K2}, options);
+
+  auto tref = t0.to(at::kFloat)
+                  .matmul(t1.t().to(at::kFloat))
+                  .matmul(t2.t().to(at::kFloat));
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, {t0, t1, t2});
+
+  auto cg_outputs = fe.runFusion({t0, t1, t2});
+
+  // relaxed check for now, err accumulation is significant.
+  TORCH_CHECK(cg_outputs[0].allclose(tref, 0.1, 0.1));
```

Comment:
Yes this case is losing quite a bit precision. 

I wasn't worrying too much about it since I cast the first matmul result to fp16 to test on mma. I guess we could revisit precision once we have fp32 and tf32 matmuls.